File: C_Sharp/Monoid/MinMonoid.cs
using System;

namespace Monoid
{
    [Serializable]
    public class MinMonoid : IMonoid<int>
    {
        private static readonly MinMonoid instance = new MinMonoid();

        protected MinMonoid()
        {
        }

        public int NeutralElement
        {
            get { return int.MaxValue; }
        }

        public int Operation(int fst, int snd)
        {
            return Math.Min(fst, snd);
        }

        public static MinMonoid Instance
        {
            get { return instance; }
        }
    }
}

File: C_plus_plus/MinMonoid.cpp
#pragma once
#include"AbstractMonoid.cpp"

class MinMonoid : public AbstractMonoid
{
public:
	
	int NeutralElement()
	{
		return -2147483648;
	}

	int Operation(int fst, int snd)
	{
		return (fst < snd ? fst : snd);
	}
};